pkg/kubelet/eviction/eviction_manager.go

```diff
@@ -196,7 +196,7 @@ func (m *managerImpl) Start(diskInfoProvider DiskInfoProvider, podFunc ActivePod
 	go func() {
 		for {
 			if evictedPods := m.synchronize(diskInfoProvider, podFunc); evictedPods != nil {
-				klog.Infof("eviction manager: pods %s evicted, waiting for pod to be cleaned up", format.Pods(evictedPods))
+				klog.InfoS("eviction manager: pods evicted, waiting for pod to be cleaned up", "pods", format.Pods(evictedPods))
```

Comment:
Please rewrite to use klog.KObj

Reply:
This comment is still applicable.

We have an interesting issue here because `klog.KObj` can't work on an array of Pods. I suggest we update `format.Pods` in order to output `[]klog.ObjectRef` instead of a single `string`. Then we can use these log entries unchanged.

We will need to do a very small refactor on the following code -- we can probably remove a lot of the code since we won't need a special `aggregatePods` with handler, instead we can just call `klog.KObj` on each pod and return the slice.

https://github.com/kubernetes/kubernetes/blob/076bd6c4010d9573d89174a1ddb17a35018c2db0/pkg/kubelet/util/format/pod.go#L60-L78

/cc @DirectXMan12

Reply:
we should probably eventually (later, not here) have a helper in klog for "turn this slice of `klog.KObj`-implementing things into `[]klog.ObjectRef`".  In the mean time, a helper in kubelet seems to be the way to go to unblock this :+1:

Reply:
> We will need to do a very small refactor on the following code -- we can probably remove a lot of the code since we won't need a special `aggregatePods` with handler, instead we can just call `klog.KObj` on each pod and return the slice.

I am interested in taking the refactor work, will set up a PR later.

Reply:
In that case I am fine with doing that in a follow-up PR and leaving anything that references `format.Pods(...)` as is for now.

Reply:
Refactor  format PR created https://github.com/kubernetes/kubernetes/pull/99799